### ISSUE TITLE
Issues with appearance, month-year and year, for date questions

### ISSUE THREAD

**Issue by yanokwa** [opendatakit/opendatakit#1169]
"I noticed with my clean Google Play version of ODK Collect (1.4.7, 1053), running on Android 5.1.1, that setting appearance::month-year or appearance::year in XLSForm for a date question does not do what it is supposed to. In both cases the app shows working month, date, and year spinners. I found this old issue https://code.google.com/p/opendatakit/issues/detail?id=873, issue 873, and it seems like this issue was supposed to have been fixed. Is there anything I am missing?"

```
<?xml version="1.0"?>
<h:html xmlns="http://www.w3.org/2002/xforms" xmlns:ev="http://www.w3.org/2001/xml-events" xmlns:h="http://www.w3.org/1999/xhtml" xmlns:jr="http://openrosa.org/javarosa" xmlns:orx="http://openrosa.org/xforms/" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <h:head>
    <h:title>dateTimeChooser</h:title>
    <model>
      <itext>
        <translation lang="Amharic">
          <text id="/dateTimeChooser/mytime:label">
            <value>Standard time</value>
          </text>
          <text id="/dateTimeChooser/year:label">
            <value>Just pick the year</value>
          </text>
          <text id="/dateTimeChooser/monthYear:label">
            <value>Just pick the month and year</value>
          </text>
          <text id="/dateTimeChooser/mydatetime:label">
            <value>Standard dateTime</value>
          </text>
          <text id="/dateTimeChooser/noCalendar:label">
            <value>No calendar should appear</value>
          </text>
          <text id="/dateTimeChooser/mydate:label">
            <value>Standard Date</value>
          </text>
        </translation>
        <translation lang="English">
          <text id="/dateTimeChooser/mytime:label">
            <value>Standard time</value>
          </text>
          <text id="/dateTimeChooser/year:label">
            <value>Just pick the year</value>
          </text>
          <text id="/dateTimeChooser/monthYear:label">
            <value>Just pick the month and year</value>
          </text>
          <text id="/dateTimeChooser/mydatetime:label">
            <value>Standard dateTime</value>
          </text>
          <text id="/dateTimeChooser/noCalendar:label">
            <value>No calendar should appear</value>
          </text>
          <text id="/dateTimeChooser/mydate:label">
            <value>Standard Date</value>
          </text>
        </translation>
      </itext>
      <instance>
        <dateTimeChooser id="dateTimeChooser">
          <mydate/>
          <mytime/>
          <mydatetime/>
          <noCalendar/>
          <monthYear/>
          <year/>
          <meta>
            <instanceID/>
          </meta>
        </dateTimeChooser>
      </instance>
      <bind nodeset="/dateTimeChooser/mydate" type="date"/>
      <bind nodeset="/dateTimeChooser/mytime" type="time"/>
      <bind nodeset="/dateTimeChooser/mydatetime" type="dateTime"/>
      <bind nodeset="/dateTimeChooser/noCalendar" type="date"/>
      <bind nodeset="/dateTimeChooser/monthYear" type="date"/>
      <bind nodeset="/dateTimeChooser/year" type="date"/>
      <bind calculate="concat('uuid:', uuid())" nodeset="/dateTimeChooser/meta/instanceID" readonly="true()" type="string"/>
    </model>
  </h:head>
  <h:body>
    <input ref="/dateTimeChooser/mydate">
      <label ref="jr:itext('/dateTimeChooser/mydate:label')"/>
    </input>
    <input ref="/dateTimeChooser/mytime">
      <label ref="jr:itext('/dateTimeChooser/mytime:label')"/>
    </input>
    <input ref="/dateTimeChooser/mydatetime">
      <label ref="jr:itext('/dateTimeChooser/mydatetime:label')"/>
    </input>
    <input appearance="no-calendar" ref="/dateTimeChooser/noCalendar">
      <label ref="jr:itext('/dateTimeChooser/noCalendar:label')"/>
    </input>
    <input appearance="month-year" ref="/dateTimeChooser/monthYear">
      <label ref="jr:itext('/dateTimeChooser/monthYear:label')"/>
    </input>
    <input appearance="year" ref="/dateTimeChooser/year">
      <label ref="jr:itext('/dateTimeChooser/year:label')"/>
    </input>
  </h:body>
</h:html>
```

**Comment by meletis** [comment]
Fixed for SurveyCTO and created a pull request:

https://github.com/opendatakit/collect/pull/6.

**Comment by mitchellsundt** [comment]
Fixed in ODK Collect 1.4.9
